lib/IRGen/GenObjC.cpp

```diff
@@ -262,39 +262,6 @@ namespace {
     ReferenceCounting getReferenceCounting() const {
       return ReferenceCounting::Bridge;
     }
-    
-    // BridgeObject exposes only null as an extra inhabitant for enum layout.
-    // Other representations are reserved for future use by the stdlib.
-    
-    bool mayHaveExtraInhabitants(IRGenModule &IGM) const override {
-      return true;
-    }
-    unsigned getFixedExtraInhabitantCount(IRGenModule &IGM) const override {
-      return 1;
-    }
-    APInt getFixedExtraInhabitantValue(IRGenModule &IGM,
-                                       unsigned bits,
-                                       unsigned index) const override {
-      return APInt(bits, 0);
-    }
-    llvm::Value *getExtraInhabitantIndex(IRGenFunction &IGF, Address src,
-                                         SILType T,
-                                         bool isOutlined) const override {
-      src = IGF.Builder.CreateBitCast(src, IGF.IGM.SizeTy->getPointerTo());
-      auto val = IGF.Builder.CreateLoad(src);
-      auto isNonzero = IGF.Builder.CreateICmpNE(val,
-                                    llvm::ConstantInt::get(IGF.IGM.SizeTy, 0));
-      // We either have extra inhabitant 0 or no extra inhabitant (-1).
-      // Conveniently, this is just a sext i1 -> i32 away.
-      return IGF.Builder.CreateSExt(isNonzero, IGF.IGM.Int32Ty);
-    }
-    void storeExtraInhabitant(IRGenFunction &IGF, llvm::Value *index,
-                              Address dest, SILType T, bool isOutlined)
-    const override {
-      // There's only one extra inhabitant, 0.
-      dest = IGF.Builder.CreateBitCast(dest, IGF.IGM.SizeTy->getPointerTo());
-      IGF.Builder.CreateStore(llvm::ConstantInt::get(IGF.IGM.SizeTy, 0), dest);
-    }
```

Comment:
Okay, I understand why this works.  As long as we're staying out of the reserved bits for tagged pointers (including BridgeObject's own notion of tagged pointers) — which the superclass implementation does — it doesn't matter if we're intruding on the BridgeObject representation because the rest of the pointer still needs to be in the valid range for objects.